bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/solcast/handler/SolcastBridgeHandler.java

```diff
@@ -0,0 +1,246 @@
+ */
+package org.openhab.binding.solarforecast.internal.solcast.handler;
+
+import static org.openhab.binding.solarforecast.internal.SolarForecastBindingConstants.*;
+
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecast;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastActions;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastProvider;
+import org.openhab.binding.solarforecast.internal.solcast.SolcastObject;
+import org.openhab.binding.solarforecast.internal.solcast.SolcastObject.QueryMode;
+import org.openhab.binding.solarforecast.internal.solcast.config.SolcastBridgeConfiguration;
+import org.openhab.binding.solarforecast.internal.utils.Utils;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.TimeSeries;
+import org.openhab.core.types.TimeSeries.Policy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolcastBridgeHandler} is a non active handler instance. It will be triggered by the bridge.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class SolcastBridgeHandler extends BaseBridgeHandler implements SolarForecastProvider, TimeZoneProvider {
+    private final Logger logger = LoggerFactory.getLogger(SolcastBridgeHandler.class);
+
+    private List<SolcastPlaneHandler> planes = new ArrayList<SolcastPlaneHandler>();
+    private Optional<SolcastBridgeConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private ZoneId timeZone;
+
+    public SolcastBridgeHandler(Bridge bridge, TimeZoneProvider tzp) {
+        super(bridge);
+        timeZone = tzp.getTimeZone();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(SolarForecastActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        SolcastBridgeConfiguration config = getConfigAs(SolcastBridgeConfiguration.class);
+        configuration = Optional.of(config);
+        if (!EMPTY.equals(config.apiKey)) {
```

Comment:
```suggestion
        if (!config.apiKey.isBlank())) {
```

You could also consider increasing readability by inverting the logic and by that decrease indentation and keep the check and corresponding status update close to each other:
```suggestion
        if (config.apiKey.isBlank())) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "@text/solarforecast.site.status.api-key-missing");
            return;
        }
```

Reply:
Correct, fixed

Reply:
It seems something went wrong here. Only the suggestion to invert the check was applied, but the logic should then update status and return.

Reply:
It's fixed